grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/AbstractComputeQuery.java

```diff
@@ -187,4 +187,22 @@ public String toString() {
                 .map(Schema.ImplicitType.HAS::getLabel)
                 .collect(Collectors.toSet());
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        AbstractComputeQuery<?> that = (AbstractComputeQuery<?>) o;
+
+        if (!graph.equals(that.graph)) return false;
+        return subTypeLabels.equals(that.subTypeLabels);
+    }
```

Comment:
This method would tell you any two path queries (or any other type of query except cluster) are the same.